### PR TITLE
fix(fedora): Recent syntax changes with DNF5

### PR DESCRIPTION
**Pull request by Angxddeep**
## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Description
There have been a recent syntax change in dnf5 which comes preinstalled with every distro based on fedora 41 or later, therefore I have updated all the setup scripts which requires **--add-repo** and replaced it with **addrepo --from-repofile="https://..."**. Also @jeevithakannan2 has added a version check so it doesn't conflict with systems that are not on dnf5.  

## Testing
Works. Although there is an issue with upstream braves repo file, I have created an issue on their GitHub page. It will start working again once they update the upstream repo file. I think this should be merged regardless as currently no script is working with dnf5 after this 3 out of 4 will start working and after brave fixes the issue all 4 will work. 

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by jeevithakannan2** [comment]
This syntax change is only for fedora 41 right?

**Comment by jeevithakannan2** [comment]
I think adding a version check is more robust way

**Comment by Angxddeep** [comment]
> I think adding a version check is more robust way

Do we really need to do that, most fedora downstream will switch to dnf5 if they haven't already done it.

**Comment by jeevithakannan2** [comment]
Are they switching to dnf5 in fedora 39 and 40 also?

**Comment by adamperkowski** [comment]
> Are they switching to dnf5 in fedora 39 and 40 also?

No idea. @Angxddeep please research this, add a version check (if applicable) and then mark this PR as ready for review. Thanks.

**Comment by lj3954** [comment]
> Are they switching to dnf5 in fedora 39 and 40 also?

No. This PR absolutely cannot be merged as is.

**Comment by Angxddeep** [comment]
I am a busy with life for a week. Ill will try to do it after that

**Comment by jeevithakannan2** [comment]
fedora 39, 40, 41 all have `dnf5`

**Comment by Angxddeep** [comment]
> fedora 39, 40 have `dnf5` package. So [64cb3c8](https://github.com/ChrisTitusTech/linutil/commit/64cb3c860be21c20d6071e7666b5f703bf745786) this might not work

They only have dnf5 is as experimental in the repos, it's not installed by default

**Comment by lj3954** [comment]
Yes, but it still could be installed; the dnf symlink will continue to point to dnf4 in that case. You would need to explicitly call dnf5 to ensure the correct program is used.

**Comment by jeevithakannan2** [comment]
dnf --version and some cuts this reduces complexity

**Comment by Angxddeep** [comment]
> Yes, but it still could be installed; the dnf symlink will continue to point to dnf4 in that case. You would need to explicitly call dnf5 to ensure the correct program is used.

I am going to be honest, fedora 39 will be EOL in a month and linutil isn't popular enough and most people will upgrade to 41 in no time. As is said I am a bit busy I think we should merge this so at least it works for people who have upgraded to F41. If you guys can submit changes that will be good too

File: core/tabs/applications-setup/Developer-tools/sublime-setup.sh [hunk 1]
```diff
@@ -23,7 +23,11 @@ installSublime() {
                 ;;
             dnf)
                 "$ESCALATION_TOOL" rpm -v --import https://download.sublimetext.com/sublimehq-rpm-pub.gpg
-                "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://download.sublimetext.com/rpm/stable/x86_64/sublime-text.repo
+                if command -v dnf5 >/dev/null 2>&1; then
+                    "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://download.sublimetext.com/rpm/dev/x86_64/sublime-text.repo
+                elif command -v dnf >/dev/null 2>&1; then
+                    "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://download.sublimetext.com/rpm/dev/x86_64/sublime-text.repo
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/Developer-tools/sublime-setup.sh:29]:
```suggestion
                dnf_version=$(dnf --version | head -n 1 | cut -d '.' -f 1)
                if [ "$dnf_version" -eq 4 ]; then
                    "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://download.sublimetext.com/rpm/dev/x86_64/sublime-text.repo
                else
                    "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://download.sublimetext.com/rpm/dev/x86_64/sublime-text.repo
```

File: core/tabs/applications-setup/browsers/brave.sh [hunk 2]
```diff
@@ -25,7 +25,11 @@ installBrave() {
                 ;;
             dnf)
                 "$ESCALATION_TOOL" "$PACKAGER" install -y dnf-plugins-core
-                "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://brave-browser-rpm-release.s3.brave.com/brave-browser.repo
+                if command -v dnf5 >/dev/null 2>&1; then
+                    "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://brave-browser-rpm-release.s3.brave.com/brave-browser.repo
+                elif command -v dnf >/dev/null 2>&1; then
+                    "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://brave-browser-rpm-release.s3.brave.com/brave-browser.repo
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browsers/brave.sh:31]:
```suggestion
                dnf_version=$(dnf --version | head -n 1 | cut -d '.' -f 1)
                if [ "$dnf_version" -eq 4 ]; then
                    "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://brave-browser-rpm-release.s3.brave.com/brave-browser.repo
                else
                    "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://brave-browser-rpm-release.s3.brave.com/brave-browser.repo
```

File: core/tabs/applications-setup/browsers/vivaldi.sh [hunk 3]
```diff
@@ -15,7 +15,11 @@ installVivaldi() {
                 ;;
             dnf)
                 "$ESCALATION_TOOL" "$PACKAGER" install -y dnf-plugins-core
-                "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://repo.vivaldi.com/stable/vivaldi-fedora.repo
+                if command -v dnf5 >/dev/null 2>&1; then
+                    "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://repo.vivaldi.com/stable/vivaldi-fedora.repo
+                elif command -v dnf >/dev/null 2>&1; then
+                    "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://repo.vivaldi.com/stable/vivaldi-fedora.repo
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browsers/vivaldi.sh:21]:
```suggestion
                dnf_version=$(dnf --version | head -n 1 | cut -d '.' -f 1)
                if [ "$dnf_version" -eq 4 ]; then
                    "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://repo.vivaldi.com/stable/vivaldi-fedora.repo
                else
                    "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://repo.vivaldi.com/stable/vivaldi-fedora.repo
```

File: core/tabs/applications-setup/docker-setup.sh [hunk 4]
```diff
@@ -28,7 +28,11 @@ install_docker() {
             ;;
         dnf)
             "$ESCALATION_TOOL" "$PACKAGER" -y install dnf-plugins-core
-            "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
+            if command -v dnf5 >/dev/null 2>&1; then
+                "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://download.docker.com/linux/fedora/docker-ce.repo
+            elif command -v dnf >/dev/null 2>&1; then
+                "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/docker-setup.sh:34]:
```suggestion
            dnf_version=$(dnf --version | head -n 1 | cut -d '.' -f 1)
            if [ "$dnf_version" -eq 4 ]; then
                "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
            else
                "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://download.docker.com/linux/fedora/docker-ce.repo
```

File: core/tabs/applications-setup/docker-setup.sh [hunk 5]
```diff
@@ -57,7 +61,11 @@ install_docker_compose() {
...
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/docker-setup.sh:67]:
```suggestion
            dnf_version=$(dnf --version | head -n 1 | cut -d '.' -f 1)
            if [ "$dnf_version" -eq 4 ]; then
                "$ESCALATION_TOOL" "$PACKAGER" config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
            else
                "$ESCALATION_TOOL" "$PACKAGER" config-manager addrepo --from-repofile=https://download.docker.com/linux/fedora/docker-ce.repo
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

**Review by jeevithakannan2 — APPROVED** [review]

**Pull request merged**